src/firefly/java/edu/caltech/ipac/worker/JobManager.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+package edu.caltech.ipac.worker;
+
+import edu.caltech.ipac.firefly.data.ServerEvent;
+import edu.caltech.ipac.firefly.server.ServerContext;
+import edu.caltech.ipac.firefly.server.events.FluxAction;
+import edu.caltech.ipac.firefly.server.events.ServerEventManager;
+import edu.caltech.ipac.firefly.server.events.WebsocketConnector;
+import edu.caltech.ipac.firefly.server.packagedata.PackagedEmail;
+import edu.caltech.ipac.firefly.server.util.Logger;
+import edu.caltech.ipac.util.AppProperties;
+import org.apache.commons.lang.text.StrBuilder;
+import org.json.simple.JSONObject;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
+
+import static edu.caltech.ipac.firefly.api.Async.getAsyncUrl;
+import static edu.caltech.ipac.firefly.data.ServerParams.EMAIL;
+import static edu.caltech.ipac.util.StringUtils.applyIfNotEmpty;
+import static edu.caltech.ipac.util.StringUtils.isEmpty;
+import static edu.caltech.ipac.worker.Job.Type.PACKAGE;
+import static edu.caltech.ipac.worker.Job.Type.SEARCH;
+import static edu.caltech.ipac.worker.JobInfo.PHASE.*;
+
+
+/**
+ * Date: 9/30/21
+ *
+ * @author loi
+ * @version : $
+ */
+public class JobManager {
+
+    private static final int KEEP_ALIVE_INTERVAL = AppProperties.getIntProperty("job.keepalive.interval", 30);    // default keepalive interval in seconds
+    private static final int WAIT_COMPLETE = AppProperties.getIntProperty("job.wait.complete", 1);                // wait for complete after submit in seconds
+
+    private static Logger.LoggerImpl LOG = Logger.getLogger();
+    private static ExecutorService packagers = Executors.newFixedThreadPool(10);
+    private static ExecutorService searches = Executors.newCachedThreadPool();
+    private static HashMap<String, JobEntry> runningJobs = new HashMap<>();
+    private static HashMap<String, JobInfo> allJobInfos = new HashMap<>();
```

Comment:
how long do we hold on to a job? I can't tell.

Reply:
JobInfo is held for the duration of the server's uptime.  However, we do cleanup the results(files), therefore, it's 7 days for zip files.  
For table searches, it's for as long as we keep the temp files, which is 1 day.  But, I am lowering it to 6hr due to the amount of local disk space needed for kubernetes in OPS.  We can talk about this.